test/boost/s3_test.cc

```diff
@@ -61,7 +61,7 @@ s3::endpoint_config_ptr make_minio_config() {
  */
 
 SEASTAR_THREAD_TEST_CASE(test_client_put_get_object) {
-    const sstring name(fmt::format("/{}/testobject-{}", tests::getenv_safe("S3_PUBLIC_BUCKET_FOR_TEST"), ::getpid()));
+    const sstring name(fmt::format("/{}/testobject-{}", tests::getenv_safe("S3_PUBLIC_BUCKET_FOR_TEST"), get_name()));
```

Comment:
> instead of using pid of current process, use the test's name for naming object used in test. less likely to cause conflicts if some test fails to perform the cleanup as expected.

Do test names differ for same test case running in different modes (dev/release/debug)?

Reply:
no. the test names do not encode the test mode, AFAICT. 

and i see your concerns -- `test.py` can drive different builds and they share the same minio server. so i still need to encode an unique id in the object name. i will amend this commit to take this into consideration. probably will have a tiny helper for this purpose.

Reply:
after a second thought, I think the “use the same name” issue is a false one, as the way how we run the test is immune to this problem. I will drop this commit from this PR.